src/mdns/mdns_avahi.hpp

```diff
@@ -266,6 +266,21 @@ class PublisherAvahi : public Publisher
      */
     otbrError UnpublishHost(const char *aName) override;
 
+    /**
+     * This method commits recent updates for a host.
+     *
+     * Some implementations require an additional step to apply recent updates for a host or its services.
+     * This method should be called after all changes with
+     * PublishHost/UnpublishHost/PublishService/UnpublishService methods are done for a given host name.
```

Comment:
Why do we need to wait for `PublishService/UnpublishService`? Does avahi forbids changes to a service entry group when a different host entry group is committed?

Reply:
I already commented on that above, but I'll try to put it a little more clear:
1) an Avahi entry group can store address and service information for a single host
2) once an entry group has been committed, its services can be modified, but they can't be removed (unpublished), new services can't be added and the IP address of the host can't be changed. In case any of the forbidden operations is needed, the group must be rebuilt from scratch (at least I couldn't find any other way to do it).
3) `CommitHost` has been added to reduce number of `reset/commit` operations in case there's a need to do several changes to the given host.

Reply:
Thanks, I get you now.

It seems you are using the same entry group for a host and all its associated services, right? What if we use separate entry groups for each services and the host? Although we manage services as a member list of host, there are no strict requirements for the service and host be in the same avahi entry group. To be more precise, service entries (TXT RR, SRV RR) and host entries (address RRs) are in different entry groups, because they have different DNS names.

I would suggest we group DNS RRs (represented by the `Service` and `Host` objects) which different DNS names in different avahi entry groups. This makes it easier to modify each service and host. Thoughts?

Reply:
I'll check if the service resolution will work if we do so.

Reply:
Thanks, I believe this should work :smiley:

Reply:
Seems to be working :) I've reworked the PR according to your suggestions.